src/Common/src/System/Numerics/Hashing/Hash.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace System.Numerics.Hashing
+{
+    internal static class Hash
+    {
+        public static int Combine(int h1, int h2)
+        {
+            return ((h1 << 5) + h1 + (h1 >> 27)) ^ h2;
+        }
```

Comment:
@VSadov Actually, it seems the current way it's written generates more efficient code. I did a disassembly dump of [this program](https://gist.github.com/jamesqo/73e8018b05c02ec3310953f45415a364) and here were the results: https://gist.github.com/7285fdea2585232610a367fd438819a9